File: src/2025-02-23-deepclaude-try/cpp/exercises/practice/dnd-character/dnd_character.cpp
#include "dnd_character.h"
#include <vector>
#include <algorithm>
#include <random>
#include <cmath>

namespace dnd_character {

int modifier(int ability_score) {
    return static_cast<int>(std::floor((ability_score - 10) / 2.0));
}

int ability() {
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<> dis(1, 6);

    std::vector<int> rolls(4);
    for (auto& roll : rolls) {
        roll = dis(gen);
    }
    std::sort(rolls.begin(), rolls.end());
    return rolls[1] + rolls[2] + rolls[3];  // Sum the three highest rolls
}

Character generate() {
    Character ch;
    ch.strength = ability();
    ch.dexterity = ability();
    ch.constitution = ability();
    ch.intelligence = ability();
    ch.wisdom = ability();
    ch.charisma = ability();

    ch.hitpoints = 10 + modifier(ch.constitution);

    return ch;
}

}  // namespace dnd_character

File: src/2025-02-23-deepclaude-try/go/exercises/practice/dnd-character/dnd_character.go
package dndcharacter

import (
	"math/rand"
	"sort"
	"time"
)

type Character struct {
	Strength     int
	Dexterity    int
	Constitution int
	Intelligence int
	Wisdom       int
	Charisma     int
	Hitpoints    int
}

// Modifier calculates the ability modifier for a given ability score
func Modifier(score int) int {
	return (score - 10) / 2
}

// Ability uses randomness to generate the score for an ability
func Ability() int {
	rolls := make([]int, 4)
	for i := 0; i < 4; i++ {
		rolls[i] = rand.Intn(6) + 1
	}
	sort.Ints(rolls)
	return rolls[1] + rolls[2] + rolls[3]
}

// GenerateCharacter creates a new Character with random scores for abilities
func GenerateCharacter() Character {
	rand.Seed(time.Now().UnixNano())
	c := Character{
		Strength:     Ability(),
		Dexterity:    Ability(),
		Constitution: Ability(),
		Intelligence: Ability(),
		Wisdom:       Ability(),
		Charisma:     Ability(),
	}
	c.Hitpoints = 10 + Modifier(c.Constitution)
	return c
}
